Sources/SafeDICore/Generators/ScopeGenerator.swift

```diff
@@ -78,8 +83,7 @@ actor ScopeGenerator {
                         return fulfillingProperty
                     }
                 }
-        ).union(propertiesToGenerate.flatMap(\.requiredReceivedProperties))
-            .subtracting(propertiesMadeAvailableByChildren)
```

Comment:
subtracting the `propertiesMadeAvailableByChildren` at the end was _wrong_, since it erased any properties we needed at this level! I re-ordered the operations above to ensure we always include the properties we need